evennia/contrib/worldloader/loader.py

```diff
@@ -0,0 +1,491 @@
+        type = 1    means ForeignKey field
+        type = 2    means ManyToManyField field, not support
+        type = -1   means field does not exist
+        """
+        
+        types = []
+        related_fields = []
+        for field_name in title:
+            type = -1
+            related_field = 0
+            
+            try:
+                # get field info
+                field = model_obj._meta.get_field(field_name)
+
+                if isinstance(field, models.ForeignKey):
+                    type = 1
+                    related_field = field.related_field
+                elif isinstance(field, models.ManyToManyField):
+                    type = 2
+                else:
+                    type = 0
+            except Exception, e:
+                logger.log_errmsg("Field error: %s" % e)
+                pass
+
+            types.append(type)
+            related_fields.append(related_field)
+        
+        # import values
+        # read next line
+        values = reader.next()
+        while values:
+            try:
+                record = {}
+                for item in zip(title, types, values, related_fields):
+                    field_name = item[0]
+                    type = item[1]
+                    value = item[2]
+                    related_field = item[3]
+                
+                    # set field values
+                    if type == 0:
+                        record[field_name] = value
+                    elif type == 1:
+                        arg = {}
+                        arg[related_field.name] = value
+                        record[field_name] = related_field.model.objects.get(**arg)
+            
+                # create new record
+                data = model_obj.objects.create(**record)
+                data.save()
+            except Exception, e:
+                logger.log_errmsg("Can't load data: %s" % e)
+
+            # read next line
+            values = reader.next()
+
+    except StopIteration:
+        # reach the end of file, pass this exception
+        pass
+
+
+
+################################################################
+#
+# These motherds set data to an object
+#
+################################################################
+
+def set_obj_data_info(obj, app, model, key):
+    """
+    Set data_info's database. It saves info to attributes of data_info category, then load these data.
+    
+    Args:
+        obj: Object in game.
+        app: (string) Db app's name.
+        model: (string) Db model's name.
+        key: (string) Key of the data info.
+    """
+    obj.attributes.add("app", app, category=DATA_INFO_CATEGORY, strattr=True)
+    obj.attributes.add("model", model, category=DATA_INFO_CATEGORY, strattr=True)
+    obj.attributes.add("key", key, category=DATA_INFO_CATEGORY, strattr=True)
+    
+    if (not app) or (not model) or (not key):
+        return True
+
+    return load_data(obj)
+
+
+def load_data(obj):
```

Comment:
The ```model``` is useful when removing an object. For example, Room01 was in table ALL_ROOMS before, but it is not in this table now. From this room's ```model``` attribute, the program know it is a room and it should not exist, so it can be removed. If there is no ```model``` attribute, the program could not know whether it should be removed.

The ```app``` is not quite unnecessary.